### PR TITLE
Events per second limits added to ossec.conf global section

### PR DESCRIPTION
**Pull request by cborla**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/12512|

## Description

The main goal of this document, is describes the configuration of a new feature implemented to manage the amount of EPS (events per second) that manager can process.

## Configuration options

```
<global>
  <limits>
    <eps>
      <maximum>500</maximum>
      <timeframe>30</timeframe>
    </eps>
  </limits>
</global>
```

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [ ] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

### CODE REVIEW

**Comment by branchnetconsulting** [comment]
I'm just trying to understand here, but what would be a real use case for configuring a Wazuh manager to artificially limit the rate of events it processes to less than its hardware is actually capable of?  I understand the point of agents self limiting their eps forwarding rate, but don't immediately appreciate how a similar rate limit would be of benefit implemented at the manager level.

**Review by TomasTurina — APPROVED** [review]

File: source/user-manual/reference/ossec-conf/global.rst [hunk 1]
```diff
@@ -489,6 +490,72 @@ Example:
...
++--------------------+-----------------------------------------------------------------+
+| **Default value**  | 0                                                               |
++--------------------+-----------------------------------------------------------------+
+| **Allowed values** | A number between 0 to 100000, Zero to disable the functionality |
++--------------------+-----------------------------------------------------------------+
+
+limits\\eps\\timeframe
+^^^^^^^^^^^^^^^^^^^^^^
+
+A positive number expressed in seconds, that indicate the time period where the events per second processed are increased and restored.
```
Comment (s-ocando) [inline source/user-manual/reference/ossec-conf/global.rst:549]:
```suggestion
A positive number expressed in seconds that indicates the time period where the events per second processed are increased and restored.
```

File: source/user-manual/api/configuration.rst [hunk 2]
```diff
@@ -277,16 +277,37 @@ access
...
 +============+======================+===============+=============================================================================================================================================================================================================================================================================================================================================+
-| enabled    | yes, true, no, false | true          | Enable or disable uploading configurations with remote commands through the Wazuh API. When this option is disabled it is not possible to upload **ossec.conf** files with the <command> option inside the :ref:`localfile tag <reference_ossec_localfile>` as well as the :ref:`wodle "command" option <wodle_command>`.                   |
+| allow      | yes, true, no, false | true          | Allow uploading configurations with remote commands through the Wazuh API. When this option is ``false`` it is not possible to upload **ossec.conf** files with the <command> option inside the :ref:`localfile tag <reference_ossec_localfile>` as well as the :ref:`wodle "command" option <wodle_command>`.                              |
```
Comment (s-ocando) [inline source/user-manual/api/configuration.rst:292]:
```suggestion
| allow      | yes, true, no, false | true          | Allow uploading configurations with remote commands through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files that contain the :ref:`wodle "command" option <wodle_command>` or the ``<command>`` option inside the :ref:`localfile tag <reference_ossec_localfile>`.                                 |
```

File: source/user-manual/api/configuration.rst [hunk 3]
```diff
@@ -277,16 +277,37 @@ access
...
+| Sub-fields | Allowed values       | Default value | Description                                                                                                                                                                                                                                                                                                                                 |
++============+======================+===============+=============================================================================================================================================================================================================================================================================================================================================+
+| allow      | yes, true, no, false | true          | Allow uploading configurations with EPS limits through the Wazuh API. When this option is ``false`` it is not possible to upload **ossec.conf** files with <limits><eps> inside the :ref:`global tag <reference_ossec_global>`.                                                                                                             |
```
Comment (s-ocando) [inline source/user-manual/api/configuration.rst:308]:
```suggestion
| allow      | yes, true, no, false | true          | Allow uploading configurations with EPS limits through the Wazuh API. Setting this option to ``false`` prevents uploading ``ossec.conf`` files with ``<limits><eps>`` inside the :ref:`global tag <reference_ossec_global>`.                                                                                                                |
```

**Review by s-ocando — COMMENTED** [review]
Please review the following suggestions.

File: source/user-manual/reference/ossec-conf/global.rst [hunk 4]
```diff
@@ -489,6 +490,72 @@ Example:
...
+
+limits\\eps\\maximum
+^^^^^^^^^^^^^^^^^^^^
+
+Maximum number of events per second allowed to be processed by decoders.
+
++--------------------+-----------------------------------------------------------------+
+| **Default value**  | 0                                                               |
++--------------------+-----------------------------------------------------------------+
+| **Allowed values** | A number between 0 to 100000, Zero to disable the functionality |
```
Comment (s-ocando) [inline source/user-manual/reference/ossec-conf/global.rst:543]:
```suggestion
| **Allowed values** | A number between 0 and 100000. Zero to disable the functionality|
```

File: source/user-manual/reference/ossec-conf/global.rst [hunk 5]
```diff
@@ -489,6 +490,72 @@ Example:
...
+
+limits\\eps\\timeframe
+^^^^^^^^^^^^^^^^^^^^^^
+
+A positive number expressed in seconds that indicates the time period where the events per second processed are increased and restored.
+
++--------------------+-------------------------------------+
+| **Default value**  | 10                                  |
++--------------------+-------------------------------------+
+| **Allowed values** | A positive number between 1 to 3600 |
```
Comment (s-ocando) [inline source/user-manual/reference/ossec-conf/global.rst:554]:
```suggestion
| **Allowed values** | A positive number between 1 and 3600|
```

**Review by s-ocando — APPROVED** [review]
LGTM

**Pull request merged**